create_primers.py

```diff
@@ -146,59 +194,64 @@ def CreateMutForOligosVarLength(seq, mutations_csv, codon_frequency_csv, primerl
     # frequent codon. This could be changed later to have random option
     primers = []
     for icodon in range(ncodons):
-        codon_inserts = []
         if icodon + 1 in mutations_dict:
             for mutation in mutations_dict[icodon + 1]:
                 max_frequency = max(back_t_dict[mutation])
                 codon_insert = back_t_dict[mutation][max_frequency]
                 i = startupper + icodon * 3
-                primer = "%s%s%s" % (seq[i - initial_flanklength : i], codon_insert, seq[i + 3 : i + 3 + initial_flanklength])
+                five_prime = seq[i - initial_flanklength : i]
+                three_prime = seq[i + 3 : i + 3 + initial_flanklength]
+                primer = f"{five_prime}{codon_insert}{three_prime}"
                 name = f"{prefix}-for-mut{icodon + 1}{mutation}"
                 primerseq = Seq(primer)
 
-                TmNN = ('%0.2f' % mt.Tm_NN(primerseq, strict=False))
+                TmNN = mt.Tm_NN(primerseq, strict=False)
                 add_3 = True
                 minus_5 = True
                 flank5 = flank3 = initial_flanklength
                 if float(TmNN) > float(maxprimertm):
```

Comment:
TmNN is now no longer a string, so I'm not sure if it's necessary to have `float()` around it, but I'd check all of this and make sure it doesn't mess up things anywhere that you're not rounding `TmNN` anymore. I could see some benefit to strictly defining how many digits `TmNN` includes, which you're not doing anymore.